packages/crab-netting/src/CrabNetting.sol

```diff
@@ -1,55 +1,992 @@
+     */
+    function toggleAuctionLive() external onlyOwner {
+        isAuctionLive = !isAuctionLive;
+    }
+
+    /**
+     * @notice set nonce to true
+     * @param _nonce the number to be set true
+     */
+    function setNonceTrue(uint256 _nonce) external {
+        nonces[msg.sender][_nonce] = true;
+    }
+
+    /**
+     * @notice set minUSDCAmount
+     * @param _amount the number to be set as minUSDC
+     */
+    function setMinUSDC(uint256 _amount) external onlyOwner {
+        minUSDCAmount = _amount;
+    }
+
+    /**
+     * @notice set minCrabAmount
+     * @param _amount the number to be set as minCrab
+     */
+    function setMinCrab(uint256 _amount) external onlyOwner {
+        minCrabAmount = _amount;
+    }
+
+    /**
+     * @notice queue USDC for deposit into crab strategy
+     * @param _amount USDC amount to deposit
+     */
+    function depositUSDC(uint256 _amount) external {
+        require(
+            _amount >= minUSDCAmount,
+            "deposit amount smaller than minimum OTC amount"
+        );
+
+        IERC20(usdc).transferFrom(msg.sender, address(this), _amount);
+
+        // update usd balance of user, add their receipt, and receipt index to user deposits index
+        usdBalance[msg.sender] = usdBalance[msg.sender] + _amount;
+        deposits.push(Receipt(msg.sender, _amount));
+        userDepositsIndex[msg.sender].push(deposits.length - 1);
+
+        emit USDCQueued(
+            msg.sender,
+            _amount,
+            usdBalance[msg.sender],
+            deposits.length - 1
+        );
+    }
+
+    /**
+     * @notice withdraw USDC from queue
+     * @param _amount USDC amount to dequeue
+     */
+    function withdrawUSDC(uint256 _amount) external {
+        require(!isAuctionLive, "auction is live");
+
+        usdBalance[msg.sender] = usdBalance[msg.sender] - _amount;
+        require(
+            usdBalance[msg.sender] >= minUSDCAmount ||
+                usdBalance[msg.sender] == 0,
+            "remaining amount smaller than minimum, consider removing full balance"
+        );
+
+        // remove that _amount the users last deposit
+        uint256 toRemove = _amount;
+        uint256 lastIndexP1 = userDepositsIndex[msg.sender].length;
+        for (uint256 i = lastIndexP1; i > 0; i--) {
+            Receipt storage r = deposits[userDepositsIndex[msg.sender][i - 1]];
+            if (r.amount > toRemove) {
+                r.amount -= toRemove;
+                toRemove = 0;
+                break;
+            } else {
+                toRemove -= r.amount;
+                delete deposits[userDepositsIndex[msg.sender][i - 1]];
+            }
+        }
+        IERC20(usdc).transfer(msg.sender, _amount);
+
+        emit USDCDeQueued(msg.sender, _amount, usdBalance[msg.sender]);
+    }
+
+    /**
+     * @notice queue Crab for withdraw from crab strategy
+     * @param _amount crab amount to withdraw
+     */
+    function queueCrabForWithdrawal(uint256 _amount) external {
```

Comment:
It's a bit confusing having 'withdraw' refer to withdrawing from the queue and withdrawing from crab. Is there better nomenclature?

Reply:
addToQueue/removeFromQueue or queueUsdc/dequeueUscd queueCrab/dequeueCrab

Reply:
done